metriq_gym/run.py

```diff
@@ -20,9 +20,34 @@
 logger.setLevel(logging.INFO)
 
 
+class QBraidSetupError(Exception):
+    pass
+
+
 def setup_device(provider_name: str, backend_name: str) -> QuantumDevice:
+    """
+    Setup QBraid quantum device for the benchmark.
+
+    Args:
+        provider_name (str): Name of the provider.
```

Comment:
[nit]: If we provide the type in the args, it's likely redundant and possibly overlooked in a refactor to update the type information here. The comments for these are a bit DRY as well. Perhaps we can put something a bit more explicit like "The metriq-gym supported provider name" or something similar?